tracee-ebpf/tracee/events_sorter_test.go

```diff
@@ -0,0 +1,176 @@
+				1: 2,
+				2: 1,
+			},
+			name: "Sorting chronological order events from multiple CPUs",
+		},
+		{
+			events: []external.Event{
+				{SMPProcessorID: 1, Timestamp: 1},
+				{SMPProcessorID: 1, Timestamp: 3},
+				{SMPProcessorID: 2, Timestamp: 2},
+			},
+			expectedCpuLastEvents: map[int]int{
+				1: 1,
+				2: 1,
+			},
+			name: "Sorting not chronological order events",
+		},
+	}
+	for _, testCase := range testCases {
+		inputChan := make(chan external.Event)
+		errChan := make(chan error)
+		newSorter := InitEventSorter(inputChan, make(chan external.Event), errChan)
+		for _, event := range testCase.events {
+			newSorter.addEvent(&event)
+		}
+		assert.Empty(t, errChan)
+		require.Len(t, newSorter.eventsQueue, len(testCase.events), testCase.name)
+		assert.True(t, sort.IsSorted(sortableEventsNodesList{newSorter.eventsQueue}), testCase.name)
+		for cpuID, expectedOffset := range testCase.expectedCpuLastEvents {
+			assert.Equal(t, expectedOffset, newSorter.cpusLastEvent[cpuID].lastEventOffset, testCase.name)
+		}
+	}
+}
+
+func TestEventsChronologicalSorter_Start(t *testing.T) {
+	testCases := []struct {
+		events []external.Event
+		name   string
+	}{{
+		events: []external.Event{
+			{SMPProcessorID: 1, Timestamp: 1},
+			{SMPProcessorID: 1, Timestamp: 2},
+			{SMPProcessorID: 1, Timestamp: 3},
+		},
+		name: "Sorting chronological order events from 1 CPU",
+	},
+		{
+			events: []external.Event{
+				{SMPProcessorID: 1, Timestamp: 1},
+				{SMPProcessorID: 2, Timestamp: 2},
+				{SMPProcessorID: 1, Timestamp: 3},
+			},
+			name: "Sorting chronological order events from multiple CPUs",
+		},
+		{
+			events: []external.Event{
+				{SMPProcessorID: 1, Timestamp: 1},
+				{SMPProcessorID: 1, Timestamp: 3},
+				{SMPProcessorID: 2, Timestamp: 2},
+			},
+			name: "Sorting not chronological order events",
+		},
+		{
+			events: []external.Event{
+				{SMPProcessorID: 1, Timestamp: 1},
+				{SMPProcessorID: 2, Timestamp: 1},
+				{SMPProcessorID: 3, Timestamp: 1},
+				{SMPProcessorID: 1, Timestamp: 3},
+				{SMPProcessorID: 2, Timestamp: 2},
+				{SMPProcessorID: 3, Timestamp: 4},
+				{SMPProcessorID: 1, Timestamp: 5},
+				{SMPProcessorID: 2, Timestamp: 3},
+				{SMPProcessorID: 3, Timestamp: 4},
+				{SMPProcessorID: 1, Timestamp: 7},
+				{SMPProcessorID: 2, Timestamp: 4},
+				{SMPProcessorID: 3, Timestamp: 4},
+				{SMPProcessorID: 1, Timestamp: 9},
+				{SMPProcessorID: 2, Timestamp: 5},
+				{SMPProcessorID: 3, Timestamp: 4},
+				{SMPProcessorID: 1, Timestamp: 8},
+				{SMPProcessorID: 2, Timestamp: 6},
+				{SMPProcessorID: 3, Timestamp: 4},
+			},
+			name: "Sorting not chronological order events complicated",
+		},
+	}
+	for _, testCase := range testCases {
+		inputChan := make(chan external.Event)
+		outputChan := make(chan external.Event)
+		errChan := make(chan error)
+		newSorter := InitEventSorter(inputChan, outputChan, errChan)
+		doneChannel := make(chan struct{})
+		go newSorter.Start(doneChannel)
```

Comment:
Looks like Start function isn't returning any type of error code. Don't we want to have it?